applications/maplab-server-node/src/maplab-server-node.cc

```diff
@@ -307,15 +332,79 @@ bool MaplabServerNode::isSubmapBlacklisted(const std::string& map_key) {
   return false;
 }
 
+bool MaplabServerNode::loadAndProcessMissingSubmaps(
+    const std::unordered_map<std::string, std::vector<std::string>>&
+        robot_to_submap_paths) {
+  std::lock_guard<std::mutex> lock(submap_callback_mutex_);
+  std::lock_guard<std::mutex> submap_queue_lock(submap_processing_queue_mutex_);
+  std::lock_guard<std::mutex> robot_lock(robot_to_mission_id_map_mutex_);
+  // We have to compare which submaps are already included in the merged map
+  // and which ones have to be loaded.
+  for (auto it = robot_to_submap_paths.begin();
+       it != robot_to_submap_paths.end(); it++) {
+    const std::string& robot_name = it->first;
+    for (std::string submap_path : it->second) {
```

Comment:
Here, we copy the `std::string` at every iteration, better use a `const ref`

Reply:
this is on purpose since it's modified by `common::simplifyPath(&submap_path)` further down